tests/contracts/test_adder_magic/src/test_adder.rs

```diff
@@ -0,0 +1,93 @@
+#![no_std]
+
+use testapi;
+
+multiversx_sc::imports!();
+
+static INIT_SUM : u32 = 5u32;
+#[multiversx_sc::contract]
+pub trait TestAdder {
+
+    #[storage_mapper("ownerAddress")]
+    fn owner_address(&self) -> SingleValueMapper<ManagedAddress>;
+
+    #[storage_mapper("adderAddress")]
+    fn adder_address(&self) -> SingleValueMapper<ManagedAddress>;
+
+    /// Create the owner account and deploy adder
+    #[init]
+    fn init(&self, code_path: ManagedBuffer) {
+        
+        // create the owner account
+        let owner = ManagedAddress::from(b"owner___________________________");
+        self.owner_address().set(&owner);
+        
+        testapi::create_account(&owner, 1, &BigUint::from(0u64));
+        
+        // register an address for the contract to be deployed
+        let adder = ManagedAddress::from(b"adder___________________________");
+        testapi::register_new_address(&owner, 1, &adder, );
+
+        // deploy the adder contract
+        let mut adder_init_args = ManagedArgBuffer::new();
+        adder_init_args.push_arg(INIT_SUM); // initial sum
+
+        // deploy a contract from `owner`
+        let adder = testapi::deploy_contract(
+                &owner,
+                5000000000000,
+                &BigUint::zero(),
+                &code_path,
+                &adder_init_args,
+            );
+
+        // save the deployed contract's address
+        self.adder_address().set(&adder);
+
+        // check the initial sum value
+        let sum_as_bytes = testapi::get_storage(&adder, &ManagedBuffer::from(b"sum")); 
+        let sum = BigUint::from(sum_as_bytes);
+        testapi::assert( sum == INIT_SUM );
+
+    }
+
+    // Make a call from 'owner' to 'adder' and check the sum value
+    #[endpoint]
+    fn test_call_add_magic(&self, value: BigUint) {
+
+        let adder = self.adder_address().get();
+
+        self.call_add(&value);
+
+        // check the sum value
+        let sum_as_bytes = testapi::get_storage(&adder, &ManagedBuffer::from(b"sum")); 
+        let sum = BigUint::from(sum_as_bytes);
+        testapi::assert( sum == value + INIT_SUM );
+        testapi::assert( sum != 987654321u32 );
```

Comment:
Shouldn't the tested contract (i.e. the adder) have this check? That would allow us to tell a story about a contract that has a bug/security issue/whatever that is hard to catch using tests. If the check is in the test function, then it's harder (but not impossible) to make an interesting story out of it.

Reply:
I see, that makes sense. I created a separate contract with magic number. See my latest commit.